src/sitemap/handler.js

```diff
@@ -393,8 +394,78 @@ export async function sitemapAuditRunner(baseURL, context) {
   };
 }
 
+async function convertToOpportunity(auditUrl, auditData, context) {
+  const { dataAccess, log } = context;
+
+  const opportunities = await dataAccess.Opportunity.allBySiteIdAndStatus(auditData.siteId, 'NEW');
+  let sitemapOpptyPagesWithIssues = opportunities.find((oppty) => oppty.getType() === 'sitemap-pages-with-issues');
+
+  const sitemapsWithPagesWithIssues = auditData.auditResult?.details?.issues
+    ? Object.keys(auditData.auditResult?.details?.issues)
+    : [];
+
+  const newData = [];
+  sitemapsWithPagesWithIssues.forEach((sitemapUrl) => {
+    const pagesWithIssues = auditData.auditResult.details.issues[sitemapUrl];
+    pagesWithIssues.forEach((page) => {
+      newData.push({
+        ...page,
+        sitemapUrl,
+      });
+    });
+  });
+
+  if (newData.length > 0) {
+    if (!sitemapOpptyPagesWithIssues) {
+      const opportunityData = {
+        siteId: auditData.siteId,
+        auditId: auditData.id,
+        runbook: 'link-to-runbook',
+        type: 'sitemap-pages-with-issues',
+        origin: 'AUTOMATON',
+        title: 'Opportunity Title',
+        description: 'Opportunity Description',
```

Comment:
this needs to replaced with a better description

Reply:
Using below description on UI
`A complete and accurate sitemap helps search engines efficiently crawl and index your website\'s pages, ensuring better visibility in search results. Fixing sitemap issues can improve the discoverability of your content.`